ouroboros-consensus/src/Ouroboros/Consensus/Util.hs

```diff
@@ -247,6 +249,21 @@ splits :: [a] -> [([a], a, [a])]
 splits []     = []
 splits (a:as) = ([], a, as) : map (\(xs, y, zs) -> (a:xs, y, zs)) (splits as)
 
+-- | Split a list given a delimiter predicate.
+--
+-- >>> split (`elem` "xy") "axbyxc"
+-- "a" :| ["b","","c"]
+--
+-- We have the laws
+--
+-- > concat (split p as) === filter (not . p) as
+-- > length (split p as) === length (filter p as) + 1
+split :: (a -> Bool) -> [a] -> NonEmpty [a]
```

Comment:
1. I added the tests for `split` in this module, but created [CAD-4730](https://input-output.atlassian.net/browse/CAD-4730) for the larger-scope follow up work.
 2. I am slightly in favor of keeping `split`: searching for it on Hoogle yields functions with identical semantics, in particular, also removing delimiters (sometimes "monomorphized" to `Text` or `ByteString`, and sometimes having signature `split' :: a -> [a] -> [[a]]` with semantics `split' a = split (a ==)`.

Reply:
Yes, but it would be rather annoying to move it back when we happen to want to use it in non-test modules. A different approach would be to merge `Test.Util.Split` into `Ouroboros.Consensus.Util`. But I don't have a strong opinion here, apart from eventually finding a better strategy of where to put these small domain-agnostic helper functions.